metricbeat/helper/kubernetes/ktest/ktest.go

```diff
@@ -19,42 +19,59 @@ package ktest
 
 import (
 	"io/ioutil"
+	"os"
+	"path/filepath"
 	"strings"
 	"testing"
 
 	p "github.com/elastic/beats/v7/metricbeat/helper/prometheus"
 	"github.com/elastic/beats/v7/metricbeat/helper/prometheus/ptest"
 )
 
-// GetTestCases Build test cases from the files and returns them
-func GetTestCases(files []string) ptest.TestCases {
+func getFiles(folder string) ([]string, error) {
+	var files []string
+	entries, err := os.ReadDir(folder)
+	if err != nil {
+		return nil, err
+	}
+	for _, e := range entries {
+		files = append(files, filepath.Join(folder, e.Name()))
+	}
+	return files, nil
+}
+
+// GetTestCases Build test cases based on the files from folder, and the expected files in the expectedFolder
+func GetTestCases(folder string, expectedFolder string) (ptest.TestCases, error) {
 	var cases ptest.TestCases
-	for i := 0; i < len(files); i++ {
+
+	files, err := getFiles(folder)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, file := range files {
 		cases = append(cases,
 			struct {
 				MetricsFile  string
 				ExpectedFile string
 			}{
-				// the metrics file is inside the parent directory, while the expected file is in the current directory
-				// Example:
-				//	Metricsfile: ../_meta/test/ksm.v2.4.2
-				//	ExpectedFile: ./_meta/test/ksm.v2.4.2.expected
-				MetricsFile:  files[i],
-				ExpectedFile: files[i][1:] + ".expected",
+				MetricsFile: file,
+				// the expected file will have the path <expectedFolder>/filename.expected
+				ExpectedFile: filepath.Join(expectedFolder, filepath.Base(file)+".expected"),
 			},
 		)
 	}
-	return cases
+	return cases, nil
 }
 
-// TestMetricsFamily
+// TestMetricsFamilyFromFiles
 // This function reads the metric files and checks if the resource fetched metrics exist in it.
 // It only checks the family metric, because if the metric doesn't have any data, we don't have a way
 // to know the labels from the file.
 // The test fails if the metric does not exist in any of the files.
 // A warning is printed if the metric is not present in all of them.
 // Nothing happens, otherwise.
-func TestMetricsFamily(t *testing.T, files []string, mapping *p.MetricsMapping) {
+func TestMetricsFamilyFromFiles(t *testing.T, files []string, mapping *p.MetricsMapping) {
 	metricsFiles := map[string][]string{}
 	for i := 0; i < len(files); i++ {
 		content, err := ioutil.ReadFile(files[i])
```

Comment:
ioutil.ReadFile(files[i] is deprected since 1.16

Can you try  io instead?
See https://stackoverflow.com/questions/75206234/for-go-ioutil-readall-ioutil-readfile-ioutil-readdir-deprecated

Reply:
Yes, fixed it.